app/client/views/home/activities/counts/activity_type/activity_type.js

```diff
@@ -2,16 +2,26 @@ Template.homeActivityCountsByActivityType.onCreated(function () {
   // Get reference to template instance
   const templateInstance = this;
 
+  let currentActivityPeriod;
   // Get home ID from template instance
   const homeId = templateInstance.data.homeId;
 
   // Set up reactive variable for chart data
   templateInstance.chartData = new ReactiveVar();
 
-  // Call method to fetch data, assigning it to reactive variable
-  Meteor.call('getHomeActivityTypeCountsLast30days', homeId, function (error, chartData) {
-    // update chart data reactive variable
-    templateInstance.chartData.set(chartData);
+  this.autorun(() => {
+    const activityPeriod = Template.currentData().activityPeriod;
+
+    // Not fetch data if period isn't changed
+    if (currentActivityPeriod !== activityPeriod) {
```

Comment:
This should be unnecessary, as autorun will only re-run when the reactive data source changes. Is this re-running when the value doesn't change?